samza-api/src/main/java/org/apache/samza/table/retry/TableRetryPolicy.java

```diff
@@ -256,4 +265,30 @@ public BackoffType getBackoffType() {
   public RetryPredicate getRetryPredicate() {
     return retryPredicate;
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<String, String> toConfig(Config jobConfig, Config tableConfig) {
+    final Gson gson = new GsonBuilder().excludeFieldsWithModifiers(Modifier.TRANSIENT, Modifier.STATIC).create();
+    return Collections.singletonMap(String.format("%s.%s", getTableId(), this.getClass().getCanonicalName()),
```

Comment:
Can we use key format that is consistent with table configuration items format?

Reply:
We can do "tables.<tableId>.<className>" to keep the similar format and avoid collision.